rfc/20241023-rfc-tracking-issues.md

```diff
@@ -0,0 +1,209 @@
+
+Overall then, this proposal aims to add a little more structure to the post-approval lifecycle for an RFC, hopefully creating a predictable pattern that the core team and the community can both rely on to understand what lifecycle phase each RFC is in, how each RFC relates to potentially-many feature request issues that inspired it, and which pull requests are related to each RFC.
+
+This proposal _does not_ significantly change the earlier stage of the authoring and approval of an RFC, except for making it explicit that it's okay for an RFC to be approved and merged even when there is no immediate plan to implement it, since we'll now have a new way to represent that situation.
+
+This proposal is also primarily discussing the process for _feature implementation_ RFCs, and not for other kinds of RFCs such as development process RFCs. Non-feature RFCs are currently far less common and so we don't have as much experience with them and thus it would be premature to try to make their process significantly more prescriptive.
+
+(The ideas in this proposal are _broadly_ based on some similar ideas from the Rust project's RFC process, although the proposed details are simpler because the OpenTofu project's current scale is considerably smaller than Rust's.)
+
+## Proposed Solution
+
+As noted above, this proposal effectively expands on what is currently the last step in the existing RFC process document:
+
+> 6. The RFC is Merged, and the Core Team creates issues in the relevant repositories to track the work required to implement the RFC.
+
+Under this proposal, immediately after merging an accepted RFC the core team creates a single "tracking issue" that represents the work to implement that RFC. The tracking RFC title is always the prefix "RFC tracker: " followed by the same title used in the RFC document itself. For now, tracking issues belong in the main `opentofu/opentofu` repository regardless of which codebases the RFC proposes to modify, though [we might change that in future](#moving-rfcs-and-their-tracking-issues-into-a-separate-repository).
+
+The tracking issue description includes the following information:
+
+- A link to the RFC text itself, and the PR(s) where the document was previously discussed.
+- Links to each of the feature request issues that were identified at the beginning of the RFC text, just to make backlinks appear on those other issues linking to the tracking issue.
```

Comment:
I had elsewhere in the doc called for assigning the tracking issue to the folks working on it, but your comment here made me realize that I left a gap: assignees give a list of people but they don't represent _what each person is responsible for_.

At the very least it seems important for the tracking issue to include the username of the "project owner", since they have some specific responsibilities that other stakeholders rely on. I feel less sure about whether it's worth getting into details like who is writing code for each feature, since that's more of a project-internal concern than something that needs to be broadly known and in practice folks' relationships to projects tend to be more complicated than just "I'm spending 100% of my time working on this". :thinking: 

Would you be comfortable with the compromise of requiring the tracking issue to mention the name of the project owner, but using the standard GitHub "assignees" for everything else for now? If we find that we're lacking some context by doing that then I'd be happy to revise that plan, but I feel that right now as a small team our relationships to projects tend to be a bit "freeform" and so I'd prefer to focus only on the one new role of "project owner" than to try to standardize _everything_ all in one step.